youtube_dl/extractor/bellesa.py

```diff
@@ -0,0 +1,109 @@
+        # nothing on the page referencing a VideoCard we cannot extract the information and
+        # thus need to raise an error
+        if 'VideoCard' not in webpage:
+            title = self._html_search_regex(
+                r'<title[^>]*>(?P<title>.+?)\s+\|\s+Bellesa',
+                webpage, 'title', default=None,
+                group='title', fatal=False)
+
+            raise ExtractorError('%s said: %s' % (self.IE_NAME, clean_html(title)), expected=True)
+
+        initial_data_raw = self._search_regex(r'window\.__INITIAL_DATA__\s+=\s+(.+?);</script>', webpage, 'initial_data')
+
+        try:
+            initial_data = json.loads(initial_data_raw)
+        except json.JSONDecodeError:
+            raise ExtractorError('%s said: cannot decode initial data', self.IE_NAME, expected=True)
+
+        video = try_get(initial_data, lambda x: x['video'])
+        if not video:
+            raise ExtractorError('%s said: initial data malformed' % self.IE_NAME, expected=True)
+
+        resolutions = try_get(video, lambda x: x['resolutions'])
+        source = try_get(video, lambda x: x['source'])
+
+        if not resolutions or not source:
+            raise ExtractorError('%s said: cannot extract playlist information from meta data' % self.IE_NAME, expected=True)
+
+        m3u8_url = 'https://s.bellesa.co/hls/v/%s/,%s,.mp4.urlset/master.m3u8' % (source, resolutions)
+        formats = self._extract_m3u8_formats(
+            m3u8_url, video_id, 'mp4',
+            entry_protocol='m3u8_native', m3u8_id='hls',
+            fatal=False)
+
+        self._sort_formats(formats)
+
+        # get from video meta data first
+        title = video.get('title')
+        if title:
+            title = title.strip()
+        else:
+            # fallback to og:title, which needs some treatment
+            title = self._og_search_title(webpage)
+            if title:
+                title = title.split('|')[0].strip()
+
+        tags = None
+        tag_string = video.get('tags')
+        if tag_string:
+            tags = [c for c in map(lambda s: s.strip(), tag_string.split(','))]
+
+        categories = None
+        if 'categories' in video:
+            categories = [c['name'] for c in video.get('categories')]
+
+        description = try_get(video, lambda x: x['description'])
+        if description:
+            description = description.strip()
+
+        return {
+            'id': video_id,
+            'title': title,
+            'thumbnail': try_get(video, lambda x: x['image']),
+            'description': description,
+            'creator': try_get(video, lambda x: x['content_provider'][0]['name']),
+            'timestamp': try_get(video, lambda x: x['posted_on']),
+            'duration': try_get(video, lambda x: x['duration']),
+            'view_count': try_get(video, lambda x: x['views']),
+            'tags': tags,
+            'categories': categories,
```

Comment:
We know that `video` is a dict, but values should be conditioned:
```suggestion
        return {
            'id': video_id,
            'title': title,
            'thumbnail': url_or_none(video.get('image')),
            'description': strip_or_none(video.get('description')) or None,
            'creator': try_get(video, lambda x: x['content_provider'][0]['name'].strip(), compat_str),
            'timestamp': int_or_none(video.get('posted_on')),
            'duration': int_or_none(video.get('duration')),
            'view_count': int_or_none(video.get('views')),
            'tags': tags or None,
            'categories': categories or None,
```